### PR TITLE
Fix mismatch between interface and model

### PR DESCRIPTION
**Pull request by olleharstedt**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | ✔️/❌

### CODE REVIEW

File: framework/db/ActiveRecord.php [hunk 1]
```diff
@@ -319,7 +319,7 @@ public function refresh()
      * For a large set of models you might consider using [[ActiveQuery::each()]] to keep memory usage within limits.
      *
      * @param array $attributes attribute values (name-value pairs) to be saved into the table
-     * @param string|array $condition the conditions that will be put in the WHERE part of the UPDATE SQL.
+     * @param string|array|null $condition the conditions that will be put in the WHERE part of the UPDATE SQL.
```
Comment (samdark) [inline framework/db/ActiveRecord.php:322]: Seems to be incorrect. Everywhere further in the calls it's string|array.

File: framework/db/ActiveRecord.php [hunk 2]
```diff
@@ -772,7 +772,7 @@ protected function deleteInternal()
      * Returns a value indicating whether the given active record is the same as the current one.
      * The comparison is made by comparing the table names and the primary key values of the two active records.
      * If one of the records [[isNewRecord|is new]] they are also considered not equal.
-     * @param ActiveRecord $record record to compare to
+     * @param ActiveRecordInterface $record record to compare to
```
Comment (samdark) [inline framework/db/ActiveRecord.php:775]: That's not correct since there's `tableName()` call and it is not in `ActiveRecordInterface`.

**Review by samdark — CHANGES_REQUESTED** [review]

**Comment by samdark** [comment]
Closing since both changes aren't correct. Thanks for contributing anyway.

**Comment by olleharstedt** [comment]
These changes are based on the Psalm analyzer. I can try and reproduce it.

**Pull request closed without merging**